ibm/service/database/resource_ibm_database.go

```diff
@@ -2432,36 +2391,102 @@ func resourceIBMDatabaseInstanceUpdate(context context.Context, d *schema.Resour
 					return diag.FromErr(fmt.Errorf(
 						"[ERROR] Error waiting for database (%s) user (%s) delete task to complete: %s", icdId, *deleteDatabaseUserOptions.Username, err))
 				}
+
+				continue
 			}
 
-			// Create New User
 			if change.New != nil {
-				userEntry := &clouddatabasesv5.User{
-					Username: core.StringPtr(change.New["name"].(string)),
-					Password: core.StringPtr(change.New["password"].(string)),
+				// No change
+				if change.Old != nil && change.Old["password"].(string) == change.New["password"].(string) && change.Old["name"].(string) == change.New["name"].(string) {
+					continue
 				}
 
-				// User Role only for ops_manager user type
-				if change.New["type"].(string) == "ops_manager" && change.New["role"].(string) != "" {
-					userEntry.Role = core.StringPtr(change.New["role"].(string))
+				oldUserExists := change.Old != nil
+
+				userUpdateCreate(change.New, instanceID, oldUserExists, meta, d)
+			}
+		}
+	}
+
+	if d.HasChange("logical_replication_slot") {
+		service := d.Get("service").(string)
+		if service != "databases-for-postgresql" {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error Logical Replication can only be set for databases-for-postgresql instances"))
+		}
+
+		cloudDatabasesClient, err := meta.(conns.ClientSession).CloudDatabasesV5()
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error getting database client settings: %s", err))
+		}
+
+		oldSlots, newSlots := d.GetChange("logical_replication_slot")
```

Comment:
Since we can't update anything, we don't need to track changes to the state. Something like this may be more straightforwardL

```go
oldList, newList := d.GetChange("logical_replication_slot")
if oldList == nil {
	oldList = new(schema.Set)
}
if newList == nil {
	newList = new(schema.Set)
}
os := oldList.(*schema.Set)
ns := newList.(*schema.Set)
remove := os.Difference(ns).List()
add := ns.Difference(os).List()

if len(add) > 0 {
	for _, slot := range add {
		addSlot := slot.(map[string]interface{})

		logicalReplicationSlot := &clouddatabasesv5.LogicalReplicationSlot{
			Name:         core.StringPtr(addSlot["name"].(string)),
			DatabaseName: core.StringPtr(addSlot["database_name"].(string)),
			PluginType:   core.StringPtr(addSlot["plugin_type"].(string)),
		}

		// ...
	}
}

if len(remove) > 0 {
	for _, slot := range remove {	
		removeSlot := slot.(map[string]interface{})
		
		// ...
	}
}
```